src/components/MessagesList/MessagesGroup/Message/MessageButtonsBar/MessageButtonsBar.vue

```diff
@@ -438,13 +450,15 @@ export default {
 				&& this.$store.getters.isActorUser()
 		},
 
-		linkToFile() {
-			if (this.isFileShare) {
-				const firstFileKey = (Object.keys(this.message.messageParameters).find(key => key.startsWith('file')))
-				return this.message.messageParameters?.[firstFileKey]?.link
-			} else {
-				return ''
-			}
+		messageFile() {
+			const firstFileKey = (Object.keys(this.message.messageParameters).find(key => key.startsWith('file')))
+			return this.message.messageParameters[firstFileKey]
+		},
+
+		linkToFileDownload() {
+			return getCurrentUser()
```

Comment:
So, it's called twice for exisitng user on each file?
I would expect this check to be called once in davUtils

Reply:
The result is already cached by `getCurrentUser()`:

https://github.com/nextcloud-libraries/nextcloud-auth/blob/67f56fa1e23b2451b4675548b650c81f8e452002/lib/user.ts#L30-L33